### PR TITLE
Analysis fixes

### PR DESCRIPTION
**Pull request by tomato42**
### Description

more fixes for the analysis module, better progress reporting, more statistics

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

- - -
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/824)

### CODE REVIEW

**Review by jogme — CHANGES_REQUESTED** [review]
I don't know your commit policy, but I would suggest to squash the commits connected to newly created files and the commits connected to them modifying them. I see two files like that: tlsfuzzer/utils/progress_report.py and tests/test_tlsfuzzer_utils_progress_report.py

File: tests/test_tlsfuzzer_utils_progress_report.py [hunk 1]
```diff
@@ -30,7 +45,68 @@ def test_all(self):
...
+class TestOperation(unittest.TestCase):
+    @mock.patch(BUILTIN_PRINT)
+    def test_with_binary_prefix(self, mock_print):
+        status = [10000, 100000, Event()]
+        params = {'prefix': 'binary', 'delay': 0.001}
+        progress = Thread(target=progress_report, args=(status,),
+                          kwargs=params)
+        progress.start()
+        while not mock_print.mock_calls:
+            pass
```
[the remark this answers is not in the document]
Comment (tomato42) [reply tests/test_tlsfuzzer_utils_progress_report.py:75]: it is waiting, that's a spinlock :)

[the remark this answers is not in the document]
Comment (jogme) [reply tests/test_tlsfuzzer_utils_progress_report.py:75]: lgtm

**Comment by tomato42** [comment]
the general policy I have for changes is _not to_ squash them, what I probably could do better job of is combining test cases with changes, but that would be rather a lot of work for little benefit

also, note that the CI already verifies that the test suite passes after each and every commit

**Pull request merged**